### PR TITLE
wip docs: Restructure docs

### PR DESCRIPTION
**Pull request by MarcoGorelli**
I think there's too many pages in the docs at the moment

I also don't really like the landing page for the api docs

It looks a bit shorter and simpler now:

image

## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [ ] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Related issues 

- Related issue # 
- Closes #

## Checklist

- [ ] Code follows style guide (ruff)
- [ ] Tests added 
- [ ] Documented the changes

### CODE REVIEW

File: docs/coming_from_pandas/overhead.md
[the remark this answers is not in the document]
Comment (MarcoGorelli) [reply docs/coming_from_pandas/overhead.md:1]: in my head I was thinking "if you're coming from pandas, rest assured, this won't make your pandas code slower", but idk, writing docs is so hard

**Comment by luke396** [comment]
Should we relocate the `typing` page or make it more accessible? 

I often forget that we have a `typing` page, even though it's linked on the homepage, as one of our main features.

(And it appears to be handwritten, not directly generated.)

{ADC37F54-1A1C-4562-8769-F8B2B28A5DB0}

**Comment by MarcoGorelli** [comment]
closing as this has been split out into other prs

**Comment by MarcoGorelli** [comment]
> Should we relocate the typing page or make it more accessible?
> 
> I often forget that we have a typing page, even though it's linked on the homepage, as one of our main features.

yup, sounds good!

**Pull request closed without merging**